src/VisualStudio/Core/Def/Implementation/PreviewPane/PreviewPaneService.cs

```diff
@@ -65,29 +65,29 @@ private Image GetSeverityIconForDiagnostic(Diagnostic diagnostic)
 
         object IPreviewPaneService.GetPreviewPane(Diagnostic diagnostic, object previewContent)
         {
-            var telemetry = diagnostic == null ? false : diagnostic.Descriptor.CustomTags.Contains(WellKnownDiagnosticTags.Telemetry);
+            var isHeaderMissing = (diagnostic == null) || diagnostic.Descriptor.CustomTags.Contains(WellKnownDiagnosticTags.Trigger);
 
-            if ((diagnostic == null) && (previewContent == null))
+            if (isHeaderMissing)
             {
-                // Bail out in cases where there is no diagnostic (which means there is nothing to put in
-                // the header section of the preview pane) as well as no preview content (i.e. no diff view).
-                return null;
-            }
+                if (previewContent == null)
+                {
+                    // Bail out in cases where there is nothing to put in the header section
+                    // of the preview pane and no preview content (i.e. no diff view) either.
+                    return null;
+                }
 
-            if ((diagnostic == null) || (diagnostic.Descriptor is TriggerDiagnosticDescriptor))
```

Comment:
once we change SimpleDiagnostic to create TriggerDiagnosticDescriptor rather than regular DiagnosticDescriptor, diagnostic should be able to round trip from diagnostic to diagnosticData without data loss, so code that checking TriggerDescriptor directly should just work.